### ISSUE TITLE
Trade's open_rate and current_rate are extremely higher than actual rate when volume = 0

### ISSUE THREAD

**Issue by mosquitotrip** [freqtrade/freqtrade#6856]
## Describe your environment

  * Operating system: Debian
  * Python Version: 3.9.12 (python -V)
  * CCXT version: 1.82.40 (pip freeze | grep ccxt)
  * Freqtrade Version: develop-8e9384e8e  (`freqtrade -V` or `docker-compose run --rm freqtrade -V` for Freqtrade running in docker)
  
Note: All issues other than enhancement requests will be closed without further comment if the above template is deleted or not filled out.

## Describe the problem:

If trade is executed when volume equals 0 then open_rate is higher than the actual current_rate (trade setting is spot/market trade)

This is from dry-run session, Haven't tested it in live (as in live it might get reported by the exchange?)

![image](https://user-images.githubusercontent.com/105578390/168904189-fc5adb18-2a80-4a3a-a0a7-201cbd06a9cf.png)
![image](https://user-images.githubusercontent.com/105578390/168905124-294b912e-9736-4aa9-86b3-69cc52ac2894.png)

I see it happens with ETF assets only

### Steps to reproduce:

  1. Generate trade-in candle with volume equal 0
  
### Observed Results:

  open_rate is higher than the actual current_rate (and visible in the plot as well)
  
  open_rate should be the rate reported by the exchange

### Relevant code exceptions or logs

```
2022-05-17 23:46:14,243 - freqtrade.wallets - INFO - Wallets synced.
2022-05-17 23:46:14,564 - freqtrade.freqtradebot - INFO - Long signal found: about create a new trade for BALBULL/USD with stake_amount: 49.14313003488 ...
2022-05-17 23:46:14,894 - freqtrade.freqtradebot - INFO - Order #dry_run_buy_1652831174.564403 was created for BALBULL/USD and status is closed.
2022-05-17 23:46:14,909 - freqtrade.wallets - INFO - Wallets synced.
2022-05-17 23:46:14,909 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: {'trade_id': 26, 'type': entry, 'buy_tag': None, 'enter_tag': None, 'exchange': 'Ftx', 'pair': 'BALBULL/USD', 'leverage': 1.0, 'direction': 'Long', 'limit': 0.0005516, 'open_rate': 0.0005516, 'order_type': 'market', 'stake_amount': 49.14756, 'stake_currency': 'USD', 'fiat_currency': 'USD', 'amount': 89100.0, 'open_date': datetime.datetime(2022, 5, 17, 23, 46, 14, 894389), 'current_rate': 0.0005515}
2022-05-17 23:46:15,053 - freqtrade.freqtradebot - INFO - Found open order for Trade(id=26, pair=BALBULL/USD, amount=89100.00000000, is_short=False, leverage=1.0, open_rate=0.00055160, open_since=2022-05-17 23:46:14)
2022-05-17 23:46:15,060 - freqtrade.freqtradebot - INFO - Fee for Trade Trade(id=26, pair=BALBULL/USD, amount=89100.00000000, is_short=False, leverage=1.0, open_rate=0.00055160, open_since=2022-05-17 23:46:14) [buy]: 0.034403292 USD - rate: 0.0007
2022-05-17 23:46:15,060 - freqtrade.persistence.trade_model - INFO - Updating trade (id=26) ...
2022-05-17 23:46:15,060 - freqtrade.persistence.trade_model - INFO - MARKET_BUY has been fulfilled for Trade(id=26, pair=BALBULL/USD, amount=89100.00000000, is_short=False, leverage=1.0, open_rate=0.00055160, open_since=2022-05-17 23:46:14).
2022-05-17 23:46:15,071 - freqtrade.wallets - INFO - Wallets synced.
2022-05-17 23:46:15,071 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: {'trade_id': 26, 'type': entry_fill, 'buy_tag': None, 'enter_tag': None, 'exchange': 'Ftx', 'pair': 'BALBULL/USD', 'leverage': 1.0, 'direction': 'Long', 'limit': 0.0005516, 'open_rate': 0.0005516, 'order_type': None, 'stake_amount': 49.14756, 'stake_currency': 'USD', 'fiat_currency': 'USD', 'amount': 89100.0, 'open_date': datetime.datetime(2022, 5, 17, 23, 46, 14, 894389), 'current_rate': 0.0005515}
2022-05-17 23:46:15,071 - freqtrade.rpc.telegram - INFO - Notification 'entry_fill' not sent.
```

**Comment by mosquitotrip** [comment]
This is how the trade appears in the UI
![image](https://user-images.githubusercontent.com/105578390/168924968-d951af14-edf5-4cf7-a876-64e2058b594f.png)
This is the exit signal, as expected
![image](https://user-images.githubusercontent.com/105578390/168925133-0fb10dbf-8875-4a3a-b694-ebb3f81df0d9.png)

**Comment by xmatthias** [comment]
We do always recommend to keep "volume > 0" condition in your strategy to avoid this.

This is actually (kindof) emulating real behaviour.
As you're using market orders (otherwise the rate would not be pushed up this high) - freqtrade tries to fill it based on available orderbook data.

If there is VERY little volume available - then freqtrade will assume a maximum slippage of 5% off of the best ask price (for buy orders).
if you entered this trade in a live scenario, you might've pushed the market even further, simply because there is no buyer/seller interest If you look at the LEOBUll/USDT chart right now - you can also observe that this is exactly what happened.

![image](https://user-images.githubusercontent.com/5024695/168958109-db378344-6a9c-4321-83d9-d84fd87728e1.png)

The orderbook is currently showing a spread of >100$ - so your market order for 0.04 LEO would - based on the current orderbook - push the price to 1.239$ (from a current spread of 1029$-1138.75$).

![image](https://user-images.githubusercontent.com/5024695/168958244-b0c8f02f-8fc8-4a5d-bb2b-4ef8d494457d.png)


As such- you should consider this a feature - as this is almost exactly how live would've performed. 
Obviously the chart will not reflect this in dry - as you didn't really push the price around (it's a simulated order only, which other market participants nor the exchange will see) - which will make it look a bit odd / surprising "on-chart".

Now the arguable point here is the 5% cutoff - your order would push the market further than the 5% - but our assumption is that once you realize this - you'll want to stay away from such pairs - and part of it in hope that volume will seep in and stop you from slipping further.

**Comment by mosquitotrip** [comment]
Thank you @xmatthias for the detailed explanations

Using volume > 0 can potentially skip (exit) signals, would SpreadFilter help better in this scenario?

makes a lot of sense as indeed the exchange will look into the orderbook:
![image](https://user-images.githubusercontent.com/105578390/168965276-855016ae-420b-4f71-a0a4-a69e4bec31c6.png)

**Comment by xmatthias** [comment]
it may skip exit signals - but the exit signal you'd get (with market orders) could push you to several losing % - as the same logic applies for sells (but in the opposite direction).

In general, i'd recommend to stay away from SUCH low-volume pairs. SpreadFilter MAY help in this particular case - but spread could also be 0.1$ - but with no volume for 100$... so that's not guaranteed to help you further.

**Comment by mosquitotrip** [comment]
Very interesting, It brings ideas like dynamic market/limit orders which can be useful for such markets instead of static options, still risky for obvious reasons.

Since I'm looking for more volatile markets, I might need to find the sweet spot with VolatilityFilter and SpreadFilter, currently, VolumeFilter is sort/filter-out logic, which can also be useful if covering a range [*/BASE] in whitelist.

But yes, indeed those markets are very risky

I can switch to limit orders, but it's a different state of mind.

While at it, a bit irrelevant to this thread - would you consider changing the hard-limit on candle history of 1000? (I noticed with binance a "&limit=1000" in the API call)

**Comment by xmatthias** [comment]
> I can switch to limit orders, but it's a different state of mind.

If you're trading such low volatility markets - best use a limit order, with `"price_side": "other"` - eventually with custom pricing, or several levels of orderbooks. That'll limit the damage you can do (to yourself and the market) by limiting the "push" up the latter into the infinite (it'll limit/control the slippage, basically).


> While at it, a bit irrelevant to this thread - would you consider changing the hard-limit on candle history of 1000? (I noticed with binance a "&limit=1000" in the API call)

1000 is the limit of the binance api - talk to their support if you'd like that changed.

You can get more than this by specifying more startup candles - but it will slow bot operations down (several calls instead of one) - so it's only recommended if the strategy REALLY needs it.

**Comment by mosquitotrip** [comment]
Interesting I'll deep dive into it.

I'll examine the results in dry/live for further optimizations.

**Comment by mosquitotrip** [comment]
SpreadFilter with > 1.25% resolved gratefully the issue (1.25% is still very high, but should be appropriate for ETFs)

![image](https://user-images.githubusercontent.com/105578390/169457018-670f9c62-db0e-46ce-8f63-440e9a3cdb0a.png)

![image](https://user-images.githubusercontent.com/105578390/169457120-3130f576-0b5e-47d9-8146-e519e8f6a9f9.png)
